types/int_test.go

```diff
@@ -109,3 +110,342 @@ func TestUint(t *testing.T) {
 	// Division-by-zero check
 	require.Panics(t, func() { i1.Div(uintmin) })
 }
+
+// Tests below uses randomness
+// Since we are using *big.Int as underlying value
+// and (U/)Int is immutable value(see TestImmutability(U/)Int)
+// it is save to use randomness in the tests
```

Comment:
I agree with the explanation that we can trust big ints to compute the expected value directly. However, we need to do a really good job of logging the random seed for each of these error messages then, and making it simple to replace the seed. 

Perhaps log in each test as `seed #%d` in the format I describe here: https://github.com/cosmos/cosmos-sdk/issues/1664

Also typo, `save -> safe`